docs/containers/reference.md

```diff
@@ -268,3 +268,147 @@ Here are all properties available for configuring `docker-run` task. All propert
+
+### Settings JSON schema
+
+There are two options for configuring each of the templates (listed below). The first is a single template which overrides the default behavior:
+
+```json
+{
+    "docker.commands.build": "docker build --rm -f \"${dockerfile}\" -t ${tag} \"${context}\""
+}
+```
+
+The second is multiple templates which will be chosen based on the `match` regular expression as well as user input. For example, two templates are given below:
+
+```json
+{
+    "docker.commands.build": [
+        {
+            "label": "Default build command",
+            "template": "docker build --rm -f \"${dockerfile}\" -t ${tag} \"${context}\""
+        },
+        {
+            "label": "Alpine-specific build command",
+            "template": "docker build -p 1234:1234 -f \"${dockerfile}\" -t ${tag} \"${context}\"",
+            "match": "alpine"
+        }
+    ]
+}
+```
+
+### Selection behavior
+
+The command template chosen to execute is selected based on the following rules:
+
+1. If no setting is configured, the default command template is chosen.
+1. If only a single template is configured (the first example above), that template is chosen.
+1. If multiple templates are configured:
+    1. Templates containing a defined `match` property are examined first. The `match` regular expression is compared against the context--for example, image name, container name, etc. All matching templates are selected. More information on the matching context is available below.
+    1. If none match the `match` property, all templates without a defined `match` property are selected.
+    1. If none match the `match` property, and there are no templates without a defined `match` property, then the default command template is chosen.
+    1. Any time that multiple templates are selected, the user will be prompted to choose between them on which to execute.
+
+### Docker Build
+
+| Configuration Setting | Default Value |
+|--|--|
+| `docker.commands.build` | `docker build --rm -f "${dockerfile}" -t ${tag} "${context}"` |
+
+Supported tokens:
+
+| Token | Description |
+| -- | -- |
+| `${dockerfile}` | The workspace-relative path of the selected `Dockerfile`. |
+| `${tag}` | The value entered/confirmed by the user upon invoking the build command. If previously built, defaults to the previously-entered value for that `Dockerfile`. |
+| `${context}` | If set, the value of the `docker.imageBuildContextPath` configuration setting. Otherwise, the workspace-relative folder in which the `Dockerfile` resides. |
+
+> NOTE: If the `docker.commands.build` setting does not contain the `${tag}` token, the user will *not* be prompted to enter/confirm a tag.
```

Comment:
Matching the style used in the VS Code docs for notes.
```suggestion
> **Note**: If the `docker.commands.build` setting does not contain the `${tag}` token, the user will *not* be prompted to enter/confirm a tag.
```

Reply:
I only just now learned about this suggested change thing, I love it!